gpkit/model.py

```diff
@@ -1,802 +0,0 @@
-# -*- coding: utf-8 -*-
-"""Module for creating Model instances.
-
-    Example
-    -------
-    >>> gp = gpkit.Model(cost, constraints, substitutions)
-
-"""
-
-import numpy as np
-
-from collections import defaultdict
-from time import time
-
-from .small_classes import Numbers, Strings
-from .nomials import MonoEQConstraint
-from .nomials import Signomial, Monomial
-from .geometric_program import GeometricProgram
-from .signomial_program import SignomialProgram
-from .solution_array import SolutionArray
-from .varkey import VarKey
-from .variables import Variable, VectorVariable
-from .posyarray import PosyArray
-from . import SignomialsEnabled
-
-from .nomial_data import NomialData
-
-from .solution_array import parse_result
-from .substitution import get_constants, separate_subs
-from .substitution import substitution
-from .small_scripts import mag, flatten, latex_num
-from .nomial_data import simplify_exps_and_cs
-from .feasibility import feasibility_model
-
-try:
-    from ipyparallel import Client
-    CLIENT = Client(timeout=0.01)
-    assert len(CLIENT) > 0
-    POOL = CLIENT[:]
-    POOL.use_dill()
-    print("Using parallel execution of sweeps on %s clients" % len(CLIENT))
-except:
-    POOL = None
-
-
-class Model(object):
-    """Symbolic representation of an optimization problem.
-
-    The Model class is used both directly to create models with constants and
-    sweeps, and indirectly inherited to create custom model classes.
-
-    Arguments
-    ---------
-    cost : Signomial (optional)
-        If this is undeclared, the Model will get its cost and constraints
-        from its "setup" method. This allows for easy inheritance.
-
-    constraints : list of Constraints (optional)
-        Defaults to an empty list.
-
-    substitutions : dict (optional)
-        This dictionary will be substituted into the problem before solving,
-        and also allows the declaration of sweeps and linked sweeps.
-
-    *args, **kwargs : Passed to the setup method for inheritance.
-
-    Attributes with side effects
-    ----------------------------
-    `program` is set during a solve
-    `solution` is set at the end of a solve
-    """
```

Comment:
We lost this nice docstring -- should a version of it get moved to `constraints/model.py`?